packages/expo-dev-client/android/src/androidTest/java/com/expo/modules/devclient/tests/DevClientLoadLocalAppTest.kt

```diff
@@ -0,0 +1,240 @@
+    """.trimIndent()
+
+private const val appURL = "http://localhost:1234"
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+internal class DevClientLoadLocalAppTest : DevLauncherKoinTest() {
+  private val rnHostCreator: ReactNativeHostCreator = { app ->
+    object : ReactNativeHost(app) {
+      override fun getUseDeveloperSupport(): Boolean {
+        return false
+      }
+
+      override fun getPackages(): List<ReactPackage> {
+        return listOf(
+          MainReactPackage(null),
+          DevLauncherPackage(),
+          DevMenuPackage(),
+          object : ReactPackage {
+            override fun createNativeModules(reactContext: ReactApplicationContext) = listOf(DevClientTestExtension(reactContext))
+            override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<View, ReactShadowNode<*>>> = emptyList()
+          })
+      }
+    }.also {
+      reactNativeHostHolder = it
+    }
+  }
+
+  private val koinDeclaration: KoinDeclaration = {
+    declareInDevLauncherScope<DevLauncherAppLoaderFactoryInterface> {
+      object : DevLauncherAppLoaderFactoryInterface, DevLauncherKoinComponent {
+        private val context: Context by inject()
+        private val appHost: ReactNativeHost by inject()
+        private val parsedManifest = StringReader(manifest).use {
+          DevLauncherManifest.fromJson(it)
+        }
+        private val controller: DevLauncherControllerInterface by inject()
+
+        override suspend fun createAppLoader(url: Uri, manifestParser: DevLauncherManifestParser): DevLauncherAppLoader {
+          Truth.assertThat(url).isEqualTo(Uri.parse(appURL))
+          return DevLauncherPublishedAppLoader(
```

Comment:
is it intentional to be using `DevLauncherPublishedAppLoader` when what we're testing seems to be a spoofed version of a locally served app? we may want to either use `DevLauncherLocalAppLoader` here, or keep `DevLauncherProductionAppLoader` but use a published/production manifest from www instead of one from expo-cli; otherwise we are testing a scenario that will never exist in the real world.